File: src/StockHawk.Service/DTOs/CreateOrderDto.cs
using System.ComponentModel.DataAnnotations;

namespace StockHawk.Service.DTOs;

public class CreateOrderDto
{
    public required string Reference { get; init; }

    [Range(0, int.MaxValue, ErrorMessage = "Please enter valid Number")]
    public required int CustomerId { get; set; }
    public DateTime OrderDate { get; set; } = DateTime.Now;

    [Range(0, double.MaxValue, ErrorMessage = "Please enter valid Number")]
    public decimal ShippingCost { get; set; }

    [Range(0, double.MaxValue, ErrorMessage = "Please enter valid Number")]
    public decimal TotalAmount { get; set; }

    [Range(0, int.MaxValue, ErrorMessage = "Please enter valid Number")]
    public required int OrderStatusId { get; set; }

    [Range(0, int.MaxValue, ErrorMessage = "Please enter valid Number")]
    public required int OrderTypeId { get; set; }
    public required ICollection<CreateOrderItemDto> OrderItems { get; init; }
}

File: src/StockHawk.Web/src/types/CreateOrderDto.ts
import { CreateOrderItemDto } from "./CreateOrderItemDto.ts";

export interface CreateOrderDto {
  reference: string;
  customerId: number;
  orderDate?: string;
  shippingCost?: number;
  totalAmount?: number;
  orderStatusId: number;
  orderTypeId: number;
  orderItems: CreateOrderItemDto[];
}
